RecoMuon/DetLayers/src/MuonME0DetLayerGeometryBuilder.cc

```diff
@@ -103,15 +106,17 @@ MuonME0DetLayerGeometryBuilder::buildLayer(int endcap,
     }
 
   }
-  LogTrace(metname) << "About to make a MuRingForwardLayer";
-  result = new MuRingForwardLayer(frontRings);
-  
-  LogTrace(metname) << "New MuRingForwardLayer with " << frontRings.size()
-                    << " and " << backRings.size()
-                    << " rings, at Z " << result->position().z()
-                    << " R1: " << result->specificSurface().innerRadius()
-                    << " R2: " << result->specificSurface().outerRadius();
   
+  LogTrace(metname) << "About to make a MuRingForwardLayer";
+  if(frontRings.size()!=0) result = new MuRingForwardLayer(frontRings);
```

Comment:
@cvuosalo - I only added the me0 layers. 
I think it isnt deleted since this object is only created at the start.
This is also the case for all the other MuonDetLayerGeometryBuilders.